File: pregunta 1/fibo1.scala
object FIBO
{
 def fibo(n:Int): Int ={
    if(n==1 || n==2) 1 else fibo(n-1)+fibo(n-2)
  }
 
 def fbn(n: Int): Int = {
    if (n == 1) {
      return 1
    }
    if (n == 2) {
      return 1
    }
    var a = 1
    var b = 1
    var c = 0
    for (i <- 3 to n) {
      c = a + b
      a = b
      b = c
    } 
     return c
 } 
 def main(args:Array[String]):Unit = {
    println("Dame un valor : ")
    val n = scala.io.StdIn.readInt()
    println("el Resultado es del recursivo : "+fibo(n)) 
    println("el Resultado es del estructurado : "+fbn(n)) 	
 }
 
}

File: pregunta 1/fibo1.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jun 13 23:58:00 2022

@author: carlos
"""

#Program to generate Fibonacci sequence recursively 
def Fibonacci(n):
    if(n==0):
        return 0
    elif(n==1):
        return 1
    else:
        return (Fibonacci(n-2)+Fibonacci(n-1))
n=int(input(" valor  Recursivo"))
for i in range(n):
    print( Fibonacci(i))

def fbn(n):
    a = 1
    b = 1
    c = 0
    
    if (n == 1):
       return 1
    if (n == 2):
      return 1

    for i in range(2,n) :
        c = a + b
        a = b
        b = c 
    return c
 
n=int(input(" valor Estructurado"))
for i in range(n):
    print( fbn(i))
    